packages/playwright-core/src/server/fetch.ts

```diff
@@ -517,7 +517,10 @@ function toHeadersArray(rawHeaders: string[]): types.HeadersArray {
 const redirectStatus = [301, 302, 303, 307, 308];
 
 function parseCookie(header: string): types.NetworkCookie | null {
-  const pairs = header.split(';').filter(s => s.trim().length > 0).map(p => p.split('=').map(s => s.trim()));
+  const pairs = header.split(';').filter(s => s.trim().length > 0).map(p => {
+    const [key, ...rest] = p.split('=');
+    return [key.trim(), rest.map(s => s.trim()).join('=')];
```

Comment:
This will break on `a=b == c`, I also liked the previous version of this code more as it didn't do unnecessary split/join, perhaps you could resurrect that and just add a check if '=' is present?

Reply:
Sorry, I just pushed a change which fixes the trim issue. I also added a test for it! Is that solution still acceptable? It's basically like the one we had before, it just handles the rest of the values a bit differently.

Reply:
I liked the version without extra `.split` calls more as the intent of the code was more clear, I'd rather you just modified it to something like the following
```js
  let key;
  let value;
  const separatorPos = p.indexOf('=');
  if (separatorPos === -1) {
    key = p.trim();
  } else {
    key = p.slice(0, separatorPos).trim();
    value = p.slice(separatorPos + 1).trim();
  }
  return [key, value];
```
but it is up to you.

Reply:
Ah indeed, it's definitely more readable! I adjusted your version and added a few comments to make clear what this is for